Small update of logger

Chupakabra0 opened the pull request:
1. Fixed correct time displaying in runtime;
2. Removed redundant things in some functions;
3. Added more macro-functions

Chupakabra0 commented on CLogger/src/Utils/Utils.cpp:
```diff
@@ -0,0 +1,38 @@
+		return str;
+	}
+	
+	std::string GetCurrentTimeString()
```
I don't know, but maybe I have to make this util-function `inline`...

AntonVoytov replied:
Could be inline

Chupakabra0 replied:
Good, so it will be done

Chupakabra0 commented on CLogger/src/Utils/Utils.cpp:
```diff
@@ -0,0 +1,38 @@
+		const auto min   = AddZero(time->tm_min);
+		const auto sec   = AddZero(time->tm_sec);
+
+		return day + "." + month + "." + year + " "
```
This date-format is hardcoded yet. Do we need to allow date-format changing?

AntonVoytov replied:
I am not sure but there is might be a function in C++/WinAPI that changing date-format. I am sure that such type of functions exists in Linux API

AntonVoytov commented on CLogger/include/Utils/Utils.h:
```diff
@@ -6,16 +6,18 @@
 /// </summary>
 namespace LogUtils
 {
+	CLOGGER_API std::string GetCurrentTimeString();
```
It would be better if these functions reduce copying strings(I mean return string by ref). I saw that you use them in macros.

Chupakabra0 replied:
OK, it will be fixed.

Chupakabra0 replied:
Stop, but what about <a href="https://docs.microsoft.com/ru-ru/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4172?f1url=%3FappId%3DDev16IDEF1%26l%3DRU-RU%26k%3Dk(C4172)%26rd%3Dtrue&view=msvc-160">C4172</a>?

AntonVoytov replied:
I mean std::string GetCurrentTimeString(); -> void/bool GetCurrentTimeString(std::string &); but your macros should be reworked then

AntonVoytov commented on CLogger/src/Utils/Utils.cpp:
```diff
@@ -0,0 +1,38 @@
+
+namespace LogUtils
+{
+	std::string AddZero(const int value)
```
Could be inline

Chupakabra0 replied:
OK

AntonVoytov commented on CLogger/include/CLogMacro/CLogMacro.h:
```diff
@@ -37,17 +38,20 @@
 
 // Returns var name
-#define GET_NAME(x) #x
+#define GET_NAME(x) #x \
```
There are a lot of macros, it will be good to reduce them, also to make public only a few of them.
For example public macros: FUNCTION_START(), FUNCTION_END(), LOG_PROD(), LOG_DEBUG(), LOG_ERROR(), LOG_WARNING()
FUNCTION_START() could use LOG_DEBUG() or LOG_TRACE() depends on log level.
We can hide a try-catch block under FUNCTION_START() and FUNCTION_END() to catch exceptions.

Chupakabra0 replied:
Interesting thought, I'll think about what can be done in the next update.

But how can I hide macros to make them "private"?

Chupakabra0 replied:
Can I hide macros via `#undef` or I must find another way?

AntonVoytov replied:
#undef should help

Chupakabra0 replied:
That's great. It'll be done.

Chupakabra0 replied:
Not great. `#undef` annihilates almost all macros `cause they try to use "undefined" macros.

Another way is to implement helper-macros in source file, but they have to be accessible for my useful header-macros at the same time.

AntonVoytov replied:
Did you try to put #undef in the end of the file with macros?

Chupakabra0 replied:
That's what I tried to do.

AntonVoytov reviewed (commented):
Read comments and think on possible solutions.

VytalyGorbatov commented on CLogger/include/Utils/Utils.h:
```diff
@@ -43,22 +17,17 @@ namespace LogUtils
-		const auto sec   = AddZero(time->tm_sec);
+		std::stringstream ss;
+		ss.imbue(std::locale());
+		ss << std::put_time(time.get(), "%c");
```
I think, it's better to use [std::strftime](https://en.cppreference.com/w/cpp/chrono/c/strftime) here.

Chupakabra0 replied:
They work the same, but `std::strftime` is C-legacy thing, so I need to create a `char` buffer before I use it. What about speed: 700-800 microseconds both variants. In this case more useful for me `std::put_time`.

Chupakabra0 commented:
Rerequest @AntonVoytov review. What was done:
1. Fixed all remarks above;
2. Added global extern vars, with which to work by macros;
3. Added `CLOG_WRITE_START_FUNCTION()` and `CLOG_WRITE_END_FUNCTION()` macros and their more efficient analogs, that hide try-catch block;
3. Fixed small bugs;
4. Fixed unworking Linux issue;
